### ISSUE TITLE
SC2046

### ISSUE THREAD

**Issue by xpz3** [koalaman/shellcheck#1522]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2046
- My shellcheck version (`shellcheck --version` or "online"): 0.4.4
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash
parent_window="$(ps --no-headers -p $(ps --no-headers -o ppid= $PPID) -o comm=)"

```

#### Here's what shellcheck currently says:

^-- SC2046: Quote this to prevent word splitting.

#### Here's what I wanted or expected to see:

I tried to remove the warning by adding quotes. But that ends in error. Can you help me with the correct way to do this? is this a false positive?

**Comment by koalaman** [comment]
This happens because your `ps` command does not just output e.g. `1` but instead <code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1</code> (with a pile of whitespace).

One way to fix this for the quoted case is to ask `ps` not to pad by setting a low field width:

    parent_window="$(ps --no-headers -p "$(ps --no-headers -o ppid:1= $PPID)" -o comm=)"

It's not really a false positive since you do in fact accidentally lose the whitespace around the output, but since you expect exactly one number surrounded by whitespace you don't care about, you can definitely justify ignoring it.
